lib/phoenix/transports/websocket.ex

```diff
@@ -2,10 +2,20 @@ defmodule Phoenix.Transports.WebSocket do
   @moduledoc false
   alias Phoenix.Socket.{V1, V2, Transport}
 
+  defmodule ConnectionErrorHandler do
+    def handle(conn, {:error, :wrong_method}) do
+      {:error, Plug.Conn.send_resp(conn, 400, "")}
+    end
+    def handle(conn, _error) do
+      {:error, Plug.Conn.send_resp(conn, 403, "")}
+    end
+  end
+
   def default_config() do
     [
       path: "/websocket",
       serializer: [{V1.JSONSerializer, "~> 1.0.0"}, {V2.JSONSerializer, "~> 2.0.0"}],
+      connection_error_handler: &__MODULE__.ConnectionErrorHandler.handle/2,
```

Comment:
We can simply call it `:error_handler`. :)

Reply:
Also, this should be a MFA, something like `{__MODULE__, :handle, []}` which you are going to invoke like this:

    {m, f, args} = config.error_handler
    apply(m, f, [conn, reason | args])

The reason we cannot use anonymous functions is because the error_handler is defined at the module level and anonymous functions do not "survive" after the module is compiled, unless they are in the `&m.f/arity` form. So it is best to use the tuple approach.

Reply:
neat, I didn't know that!